### DISCUSSION TITLE
Include program in crash report

### DISCUSSION THREAD

**Discussion by jaschdoc** [flix/flix#8541] [Ideas]
It would be nice the crash reports contained the program that caused the crash. Hopefully, this is not too difficult with the `DocAst`. That way it is easier for us to try to find a minimal program and when it happens for someone else, they can just send in the crash report as is.
Another solution could be to store the parsed AST in `Flix.scala` and then pretty print / unparse that on crash. It all depends on what is most relevant to us.
To avoid printing huge programs we could also just print the contents of the file that was most recently updated. There are probably other tricks we can do to avoid long output.

**Comment by magnus-madsen** [comment]
There are some privacy concerns though.

**Comment by jaschdoc** [comment]
True, maybe this could be a flag that we run without per default so it's opt-in?

**Comment by JonathanStarup** [comment]
Related discussion  #8522

I don't think we should use prettyprinting - that would require correct pretty printing, no desugaring in the frontend (currently, desugaring starts even in the weeder), and it doesn't work for parser errors.

**Comment by jaschdoc** [comment]
Just to be clear, I don't mean pretty output from pretty printing. I just thought that printing some amount of the program (with errors) where the crash location occurred would be nice to include.

**Reply by magnus-madsen** [reply]
Unfortunately, often when a crash occurs we do not have the source location :/

**Reply by jaschdoc** [reply]
Sometimes we do and we could try to include it then
